src/vision.cpp

```diff
@@ -29,29 +41,35 @@ Vision::Vision(ros::NodeHandle nh_camera, ros::NodeHandle nh_private)
 
 Vision::~Vision()
 {
+  quit();
+}
+
+rclcpp::node_interfaces::NodeBaseInterface::SharedPtr Vision::getNodeBaseInterface()
+{
+  return node_->get_node_base_interface();
 }
 
 bool Vision::configure()
 {
   std::string streamConfig_rosparam = "";
   bool bStreamConfigDefined = false;
-  char* streamConfig_env = NULL;
-
-  bStreamConfigDefined = nh_private_.getParam("stream_config", streamConfig_rosparam);
 
+  node_->declare_parameter<std::string>("stream_config");
```

Comment:
I see "stream_config" written out 3 times in a row here, might not be a bad idea to create a const variable for it in the empty namespace